src/server/routes.ts

```diff
@@ -17,14 +17,30 @@ export default router;
 
 async function searchProducts(name: string) {
   const response = await fetch(
-    `https://www.idealo.de/preisvergleich/MainSearchProductCategory.html?q=${name}`
+    `https://www.idealo.de/preisvergleich/MainSearchProductCategory.html?q=${name}`,
+    {
+      headers: {
+        'accept-language': 'de-DE',
+      },
+    }
   );
-  const rawHtml = await response.text();
 
-  const parser = new domParser();
-  const dom = parser.parseFromString(rawHtml);
-  const price = dom.getElementsByClassName('myofferList-item-priceMin');
-  const product = { name: name, producer: 'tba', price: price };
-  const products = [product];
-  return products;
+  const rawHTML = await response.text();
+  const $ = cheerio.load(rawHTML);
+  const offerListItems = $('.offerList-item');
+  const products: {
+    title: string;
+    price: string;
+  }[] = [];
+  offerListItems.each((_i, offerListItem) => {
+    const title = $(offerListItem)
+      .find('.offerList-item-description-title')
+      .text();
+    const price = $(offerListItem).find('.offerList-item-priceMin').text();
```

Comment:
if you don't get data you can try adding this:

.find('.offerList-item-priceWrapper > .offerList-item-pricemin') as a cheerio selector

Reply:
Is cheerio comparing these?

Reply:
for me in some cases I need to "search" from a higher dom level. but maybe this only works for me 🤷‍♂️